protocol/communities/manager.go

```diff
@@ -1530,6 +1609,164 @@ func (m *Manager) IsSeedingHistoryArchiveTorrent(communityID types.HexBytes) boo
 	return ok && torrent.Seeding()
 }
 
+func (m *Manager) DownloadHistoryArchivesByMagnetlink(communityID types.HexBytes, magnetlink string) ([]string, error) {
+
+	id := communityID.String()
+	ml, err := metainfo.ParseMagnetUri(magnetlink)
+	if err != nil {
+		return nil, err
+	}
+
+	m.archiveLogger.Debug("adding torrent via magnetlink for community", zap.String("id", id), zap.String("magnetlink", magnetlink))
+	torrent, err := m.torrentClient.AddMagnet(magnetlink)
+	if err != nil {
+		return nil, err
+	}
+	m.torrentTasks[id] = ml.InfoHash
+
+	select {
+	case <-time.After(20 * time.Second):
+		return nil, errors.New("torrent has timed out")
+	case <-torrent.GotInfo():
```

Comment:
@cammellos I've added a time out of 20 seconds here in case `torrent.GotInfo()` doesn't emit anything